src/coffee/directives/api/utils/model-key.coffee

```diff
@@ -36,7 +36,7 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
         #compare the rest of the properties that are being watched by scope
         without = _.without(@interface.scopeKeys, 'coords')
         isEqual = _.every without, (k) =>
-          @scopeOrModelVal(scope[k], scope, model1) == @scopeOrModelVal(scope[k], scope, model2)
+          _.isEqual(@scopeOrModelVal(scope[k], scope, model1), @scopeOrModelVal(scope[k], scope, model2))
```

Comment:
This will affect speed for sure. The depth of what is watched should be optional. I know the 4.X branch of lodash is much much faster on this. But this will have a negative effect on users stuck on old lodash,.